django/contrib/admin/views/main.py

```diff
@@ -52,6 +53,9 @@ def __init__(self, request, model, list_display, list_display_links,
         self.list_max_show_all = list_max_show_all
         self.model_admin = model_admin
         self.preserved_filters = model_admin.get_preserved_filters(request)
+        if not orderable_by:
```

Comment:
This doesn't seem correct -- we will call `get_orderable_by` redundantly if it's set to an empty tuple. I don't see a reason not to make `orderable_by` a positional argument and updating the tests.